File: transmision digital/matlab/2/simulación/onda3.py
import numpy as np
import matplotlib.pyplot as plt

# Definir la función escalón unitario
def u(t):
    return np.where(t >= 0, 1, 0)

# Rango de tiempo
t = np.arange(-2, 10.01, 0.01)

# Definir la función por partes
y = 0.5*(-np.cos(0.5*np.pi*t)+1)*(u(t)-u(t-4))+0.5*(np.cos(0.5*np.pi*t)-1)*(u(t-4)-u(t-8))

# Graficar
plt.plot(t, y, 'r', linewidth=2)
plt.axis([-1, 8, -3, 3])
plt.grid(True)
plt.xlabel('Tiempo (t)')
plt.ylabel('y(t)')
plt.title('Gráfica de la función definida por partes')
plt.show()

File: transmision digital/matlab/2/onda3.m
clear;
close all;
clc;

% Definir la función escalón unitario
u = @(t) (t >= 0); 

% Rango de tiempo
x = -1:0.01:10; 
t = x*4;
% Definir la función por partes
%y = (-cos(0.5*pi*t)+1).*(u(t)-u(t-4))+(cos(0.5*pi*t)-1).*(u(t-4)-u(t-8));
y = 0.5*(-cos(0.5*pi*t)+1).*(u(t)-u(t-4))+0.5*(cos(0.5*pi*t)-1).*(u(t-4)-u(t-8));
g = y*2;
% Graficar la función
plot(t, g, 'r', 'LineWidth', 2);
hold on;
%plot(t, g, 'b', 'LineWidth', 2);
axis([0 8 -3 3]); % Límites de los ejes
hold off;
grid on;


xlabel('Tiempo (t)');
ylabel('y(t)');
title('Gráfica de la función definida por partes');